docs/docs/icicle/introduction.md

```diff
@@ -8,31 +8,37 @@ This guide is oriented towards developers who want to start writing code with th
+ICICLE Core compiles into a static library. This library can be used with our official Golang and Rust wrappers or linked with your C++ project. You can also implement a wrapper for it in any other language.
 
-Based on this dependency architecture, the ICICLE repository has three main sections, each of which is independent from the other.
+Based on this dependency architecture, the ICICLE repository has three main sections:
 
-- ICICLE core
-- ICICLE Rust bindings
-- ICICLE Golang bindings
+- [ICICLE Core](#icicle-core)
+- [ICICLE Rust bindings](#icicle-rust-and-golang-bindings)
+- [ICICLE Golang bindings](#icicle-rust-and-golang-bindings)
 
 ### ICICLE Core
 
-[ICICLE core](https://github.com/ingonyama-zk/icicle/tree/main/icicle) contains all the low level CUDA code implementing primitives such as [points](https://github.com/ingonyama-zk/icicle/tree/main/icicle/primitives) and [MSM](https://github.com/ingonyama-zk/icicle/tree/main/icicle/appUtils/msm). There also exists higher level C++ wrappers to expose the low level CUDA primitives ([example](https://github.com/ingonyama-zk/icicle/blob/c1a32a9879a7612916e05aa3098f76144de4109e/icicle/appUtils/msm/msm.cu#L1)).
+[ICICLE Core](/icicle/core) is a library that directly works with GPU by defining CUDA kernels and algorithms that invoke them. It contains code for [fast field arithmetic](https://github.com/ingonyama-zk/icicle/tree/main/icicle/include/field/field.cuh), cryptographic primitives used in ZK such as [NTT](https://github.com/ingonyama-zk/icicle/tree/main/icicle/src/ntt/), [MSM](https://github.com/ingonyama-zk/icicle/tree/main/icicle/src/msm/), [Poseidon Hash](https://github.com/ingonyama-zk/icicle/tree/main/icicle/src/poseidon/), [Polynomials](https://github.com/ingonyama-zk/icicle/tree/main/icicle/src/polynomials/) and others.
 
-ICICLE Core would typically be compiled into a static library and used in a third party language such as Rust or Golang.
+ICICLE Core would typically be compiled into a static library and either used in a third party language such as Rust or Golang, or linked with your own C++ project.
 
 ### ICICLE Rust and Golang bindings
 
-- [ICICLE Rust bindings](https://github.com/ingonyama-zk/icicle/tree/main/wrappers/rust)
-- [ICICLE Golang bindings](https://github.com/ingonyama-zk/icicle/tree/main/goicicle)
+- [ICICLE Rust bindings](/icicle/rust-bindings)
+- [ICICLE Golang bindings](/icicle/golang-bindings)
 
 These bindings allow you to easily use ICICLE in a Rust or Golang project. Setting up Golang bindings requires a bit of extra steps compared to the Rust bindings which utilize the `cargo build` tool.
 
 ## Running ICICLE
 
 This guide assumes that you have a Linux or Windows machine with an Nvidia GPU installed. If you don't have access to an Nvidia GPU you can access one for free on [Google Colab](https://colab.google/).
 
+:::info note
+
+ICICLE can only run on Linux or Windows. **MacOS is not supported**.
```

Comment:
Maybe it's better to note that ICICLE is running on CUDA at this time which is supported only on Nvidia GPUs, therefore ICICLE cannot run on MacOS

Reply:
this is mentioned in the next section. The idea of this note is to send a short and understandable message that we don't support MacOS (and probably - never will)

Reply:
even if we decide to use metal - this will most definitely be a different library